### ISSUE TITLE
bug with using variable

### ISSUE THREAD

**Issue by sergeych** [p0deje/yard-doctest#6]
<pre>
# @example Hash/Array keep the structure
#   hash = { 'foo' => 'bar' }
#   Boss.pack [hash, hash]             #=> "\x16\x0F\efoo\ebar\x15"
#   src = Boss.unpack("\x16\x0F\efoo\ebar\x15")
#   src[0]['foo'] == 'bar'
#   #=> true
#   src[1]['foo'] = 'bar'
#   #=> true
</pre>

yard doctest:
<pre>
1) Failure:
Boss#test_0001_Hash/Array keep the structure [/Users/sergeych/dev/boss_protocol/lib/boss-protocol.rb:90]:
--- expected
+++ actual
@@ -1 +1 @@
-true
+"#<NameError: undefined local variable or method `src' for Boss:Module>"
</pre>
It is caused by the `src[1]` line! If I remove it, test works as expected.

**Comment by p0deje** [comment]
You have a typo in that line. It should be `==` not `=`.

**Comment by sergeych** [comment]
Sorry I put the wrong sample. and foget to escape <>. With '==' it gives the same error I'm trying to report and it is about m:
<pre>
  1) Failure:
Boss#test_0001_Hash/Array keep the structure [/Users/sergeych/dev/boss_protocol/lib/boss-protocol.rb:99]:
--- expected
+++ actual
@@ -1 +1 @@
-true
+"#&lt;NameError: undefined local variable or method `src' for Boss:Module&gt;"
</pre>
Please note that the problem is, executing the second line (src[1]) the src variable itself does not exist in the scope dove the test is being executed.

**Comment by p0deje** [comment]
I cannot reproduce the failure. I've added a simple test:

```cucumber
  Scenario: names test with example title when it's present
    Given a file named "doctest_helper.rb" with:
      """
      require 'app/app'
      """
    And a file named "app/app.rb" with:
      """
      # @example
      #   hash = {'foo' => 'bar'}
      #   src = hash
      #   src['foo'] == 'bar'
      #   #=> true
      #   src['foo'] == 'bar'
      #   #=> true
      def foo
      end
      """
    When I run `bundle exec yard doctest -v`
    Then the output should contain "1 runs, 2 assertions, 0 failures, 0 errors, 0 skips"
```

This scenario passes, so if you can provide a failure test case, that would be great.

Also, make sure you use the latest version of yard-doctest.

**Comment by sergeych** [comment]
try:
<pre>
Feature: test doctest

Scenario: names test with example title when it's present
    Given a file named "doctest_helper.rb" with:
      """
      require 'app/app'
      """
    And a file named "app/app.rb" with:
      """
      module Test
      # @example
      #   hash = {'foo' => 'bar'}
      #   src = hash
      #   src['foo'] == 'bar'
      #   #=> true
      #   src['foo'] == 'bar'
      #   #=> true
      def foo
      end
      end
      """
    When I run `bundle exec yard doctest -v`
    Then the output should contain "1 runs, 2 assertions, 0 failures, 0 errors, 0 skips"
</pre>

**Comment by p0deje** [comment]
Aha, I can reproduce it, thank you!

**Comment by p0deje** [comment]
This should be fixed in 0.1.10, please give it a shot!
